File: PYTHON/nth_Fibonacci_Number.py

class Solution:
    def nthFibonacci(self, n : int) -> int:
        # code here
        arr = []
        arr.append(0)
        arr.append(1)
        for i in range(2,n+1):
            arr.append((arr[i-1]+arr[i-2])%1000000007)
        return arr[n]%1000000007
        

File: JAVA/nth_Fibonacci_Number.java
/**
 * nth_Fibonacci_Number
 */
public class nth_Fibonacci_Number {

    class Solution {
        static int nthFibonacci(int n){
            int arr[] = new int[n+1];
            arr[0]=0;
            arr[1]=1;
            
            for(int i = 2;i<n+1;i++){
                arr[i] = (arr[i-1]+arr[i-2])%1000000007;
            }
            return arr[n];
        }
    }
}